### PR TITLE
Fix comps reposync

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Fix downloading comps files by matching type in repomd.xml and using globs on the directory.
The file names change all the time.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- unit tests added

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/15044
Tracks https://github.com/SUSE/spacewalk/pull/15056

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"  
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: backend/satellite_tools/repo_plugins/yum_src.py [hunk 1]
```diff
@@ -704,12 +704,25 @@ def _retrieve_md_path(self, tag):
...
+                if sub_item.tag.endswith("location"):
+                    return sub_item.attrib.get("href")
+
+        repomd = open(repomd_path, 'rb')
+        path = None
+        for _event, elem in etree.iterparse(repomd):
+            if elem.tag.endswith("data") and elem.attrib.get("type") == tag:
+                path = get_location(elem)
+                break
+        repomd.close()
```
Comment (meaksh) [inline backend/satellite_tools/repo_plugins/yum_src.py:722]:
```suggestion
        def get_location_from_xml_element(data_item):
            for sub_item in data_item:
                if sub_item.tag.endswith("location"):
                    return sub_item.attrib.get("href")
                    
        path = None
        with open(repomd_path, 'rb') as repomd:
            for _, elem in etree.iterparse(repomd):
                if elem.tag.endswith("data") and elem.attrib.get("type") == tag:
                    path = get_location_from_xml_element(elem)
                    break
```

**Review by meaksh — COMMENTED** [review]
I've added a minor code refactor (cosmetic) which I think increases readability.

Other than that, LGTM.

**Review by hustodemon — COMMENTED** [review]
Left 2 instances of 1 question. Otherwise the code looks good to me!

One more note though: I noticed we have a test for this file: `test_yum_src.py`. Would it be feasible to test the new behavior (possibly on real xml data extracted from existing repos)? That'd be a great fixture for the future. Is this be possible with the current testing library @meaksh ?

File: backend/satellite_tools/repo_plugins/yum_src.py [hunk 2]
```diff
@@ -704,12 +704,24 @@ def _retrieve_md_path(self, tag):
...
-            if _md_files:
-                return _md_files[0]
-        return None
+        repomd_path = os.path.join(_repodata_path, 'repomd.xml')
+        if tag == 'repomd':
+            return repomd_path
+
+        def get_location_from_xml_element(data_item):
+            for sub_item in data_item:
+                if sub_item.tag.endswith("location"):
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply backend/satellite_tools/repo_plugins/yum_src.py:713]: We use endswith to be independent of the namespace which sometimes exists on element tags

File: backend/satellite_tools/repo_plugins/yum_src.py [hunk 3]
```diff
@@ -704,12 +704,24 @@ def _retrieve_md_path(self, tag):
...
+                    return sub_item.attrib.get("href")
+                    
+        path = None
+        with open(repomd_path, 'rb') as repomd:
+            for _, elem in etree.iterparse(repomd):
+                if elem.tag.endswith("data") and elem.attrib.get("type") == tag:
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply backend/satellite_tools/repo_plugins/yum_src.py:719]: We use endswith to be independent of the namespace which sometimes exists on element tags

**Review by hustodemon — APPROVED** [review]
LGTM, thanks for adding the tests @mcalmer  :+1: !

**Review by meaksh — APPROVED** [review]

**Review by meaksh — APPROVED** [review]

**Pull request merged**